cpp/velox/shuffle/VeloxSplitter.h

```diff
@@ -0,0 +1,370 @@
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "velox/type/Type.h"
+#include "velox/vector/ComplexVector.h"
+#include "velox/vector/FlatVector.h"
+
+#include <arrow/filesystem/filesystem.h>
+#include <arrow/filesystem/localfs.h>
+#include <arrow/io/api.h>
+#include <arrow/ipc/writer.h>
+#include <arrow/memory_pool.h>
+#include <arrow/record_batch.h>
+#include <arrow/type.h>
+#include <arrow/type_traits.h>
+#include <arrow/util/bit_util.h>
+#include <arrow/util/checked_cast.h>
+#include "arrow/array/builder_base.h"
+
+#include "arrow/array/util.h"
+#include "arrow/result.h"
+
+#include "operators/shuffle/type.h"
+#include "operators/shuffle/utils.h"
+
+namespace gluten {
+
+class VeloxSplitter {
+  enum { VALIDITY_BUFFER_INDEX = 0, OFFSET_BUFFER_INDEX = 1, VALUE_BUFFER_INEDX = 2 };
+
+ public:
+  struct BinaryBuff {
+    BinaryBuff(uint8_t* value, uint8_t* offset, uint64_t value_capacity, uint64_t value_offset)
+        : value_ptr(value), offset_ptr(offset), value_capacity(value_capacity), value_offset(value_offset) {}
+
+    BinaryBuff(uint8_t* value, uint8_t* offset, uint64_t value_capacity)
+        : BinaryBuff(value, offset, value_capacity, 0) {}
+
+    BinaryBuff() : BinaryBuff(nullptr, nullptr, 0, 0) {}
+
+    uint8_t* value_ptr;
+    uint8_t* offset_ptr;
+    uint64_t value_capacity;
+    uint64_t value_offset;
+  };
+
+  template <typename SPLITTER>
+  static std::shared_ptr<SPLITTER> Create(uint32_t num_partitions, const SplitOptions& options) {
+    return std::make_shared<SPLITTER>(num_partitions, options);
+  }
+
+  static arrow::Result<std::shared_ptr<VeloxSplitter>>
+  Make(const std::string& name, uint32_t num_partitions, SplitOptions options = SplitOptions::Defaults());
+
+  virtual const std::shared_ptr<arrow::Schema>& input_schema() const {
```

Comment:
Can we get arrow Schema here?

Reply:
I kept this interface here,  I will delete it after I add the code for `JNI` and find it is useless. OK?